specification/datadog/resource-manager/Microsoft.Datadog/stable/2022-08-01/datadog.json

```diff
@@ -0,0 +1,2013 @@
+      "description": "Flag specifying if the resource monitoring is enabled or disabled.",
+      "enum": [
+        "Enabled",
+        "Disabled"
+      ],
+      "default": "Enabled",
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "MonitoringStatus"
+      }
+    },
+    "MarketplaceSubscriptionStatus": {
+      "description": "Flag specifying the Marketplace Subscription Status of the resource. If payment is not made in time, the resource will go in Suspended state.",
+      "enum": [
+        "Provisioning",
+        "Active",
+        "Suspended",
+        "Unsubscribed"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "MarketplaceSubscriptionStatus"
+      },
+      "readOnly": true
+    },
+    "DatadogOrganizationProperties": {
+      "description": "Datadog organization properties",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the Datadog organization.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "id": {
+          "description": "Id of the Datadog organization.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "linkingAuthCode": {
+          "description": "The auth code used to linking to an existing datadog organization.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create"
+          ],
+          "x-ms-secret": true
+        },
+        "linkingClientId": {
+          "description": "The client_id from an existing in exchange for an auth token to link organization.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create"
+          ],
+          "x-ms-secret": true
+        },
+        "redirectUri": {
+          "description": "The redirect uri for linking.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "apiKey": {
+          "description": "Api key associated to the Datadog organization.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create"
+          ],
+          "x-ms-secret": true
+        },
+        "applicationKey": {
+          "description": "Application key associated to the Datadog organization.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create"
+          ],
+          "x-ms-secret": true
+        },
+        "enterpriseAppId": {
+          "description": "The Id of the Enterprise App used for Single sign on.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "cspm": {
+          "description": "The configuration which describes the state of cloud security posture management",
+          "type": "boolean"
```

Comment:
>"type": "boolean [](http://example.com/codeflow?start=10&length=16)

Consider to use enum for future extensibility. 

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=enum](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=enum)